src/Microsoft.DotNet.Cli.Utils/CommandResolution/ProjectDependenciesCommandResolver.cs

```diff
@@ -136,13 +107,15 @@ public CommandSpec Resolve(CommandResolverArguments commandResolverArguments)
                         depsFilePath);
         }
 
-        private IEnumerable<LockFilePackageLibrary> GetAllDependencyLibraries(
-            ProjectContext projectContext)
+        private LockFileTargetLibrary GetToolLibraryForContext(
+            ProjectContext projectContext, string commandName)
         {
-            return projectContext.LibraryManager.GetLibraries()
-                .Where(l => l.GetType() == typeof(PackageDescription))
-                .Select(l => l as PackageDescription)
-                .Select(p => p.Library);
+            var toolLibrary = projectContext.LockFile.Targets
```

Comment:
Calling out that this is essentially bypassing the framework/rid inherent in the projectContext to search for the proper target. 

A cleaner approach would be to iterate through the possible project contexts at a higher level and call this multiple times, or pass in the project directory and parse the lock file directly.